### ISSUE TITLE
GTK4 support

### ISSUE THREAD

**Issue by JohnOberhauser** [themix-project/oomox-gtk-theme#75]
Arch is using gnome 40 with gtk4 support now.  It would be awesome if the oomox theme supported gtk4.

**Comment by JohnOberhauser** [comment]
yeah, I ended up using hacking materia to look similar-ish to oomox/numix.  It ended up looking pretty good

**Comment by actionless** [comment]
not good enough for sharing the result though?

**Comment by actionless** [comment]
considering all weird things happening with GTK4:

i personally dont see much point investing time into it until theming api will became 1) documented; and 2) "changelog-able"

but as i mentioned before, PRs still will be accepted

**Comment by yavko** [comment]
There is a pretty easy way to theme libadwaita apps, though IMO supporting gtk4 for non libadwaita apps or other things would be great. I think there is more documentation on it though I'm not sure.

**Comment by actionless** [comment]
@yavko can't wait for that cool PR from you based on their robust documentation :D

**Comment by yavko** [comment]
> @yavko can't wait for that cool PR from you based on their robust documentation :D

Even though I don't use oomox-gtk, but I'll do it! But idk how long it'll take 💀, so no guarantees

**Comment by actionless** [comment]
btw i actually tested latest versions of oomox-gtk3-theme and gtk4-widget-factory and it's aparently not so much fixes to do visually (at least from the first glance, mb in other places it all worse), but there are lots of theme parsing errors, due to some changes in gtk not clearly described in CHANGELOG.txt:


gtk3:
![2022-04-05--1649152847_1920x1080_scrot](https://user-images.githubusercontent.com/1655669/161730219-e299e046-6b3b-4e37-bfe9-374085285fbf.png)



gtk4:
![2022-04-05--1649152883_1920x1080_scrot](https://user-images.githubusercontent.com/1655669/161730243-aefefb03-0ef7-49b1-ac5f-349e79021f5c.png)

**Comment by actionless** [comment]
another option is to switch to libadwaita's css for gtk4, because by structure it's similar to materia but doesnt have any image assets to render - it's quite easy to use in Themix and visual changes from Oomox theme are notable but not really crucial:

![2022-06-02--1654198213_1688x969_scrot](https://user-images.githubusercontent.com/1655669/171723170-1ebf3b5a-bea6-46fe-97e8-696b75728a00.png)

**Comment by yavko** [comment]
I finally have time to try porting it myself, though as I said before I can't guarantee I will finish it at all lol

**Comment by niksingh710** [comment]
[wpgtk issue](https://github.com/deviantfero/wpgtk/issues/264#issuecomment-1419344388)
Here in this issue wpgtk dev created a theme linea that works with gtk4.
maybe that helps here to get gtk4 support

**Comment by actionless** [comment]
@niksingh710 the problem is not "how to create any gtk4 theme" (https://github.com/themix-project/themix-plugin-base16/tree/master/templates_extra/gtk4-oodwaita/templates), the problem that no one is in sane mind to spend their own time to rewrite existing oomox theme to randomly changed (and what's more important always changING) gtk theming api created by distalented gnome team, whos bigger project goal is wasting as much ibm money as possible without delivering anything useful. (because otherwise the gnome project will be finished and they will be sent doing some boring servering software)

**Comment by actionless** [comment]
however, since wpgtk theme support both gtk3 and gtk4 with the same style - feel free adding themix plugin for that theme: https://github.com/themix-project/themix-gui/tree/master/plugins

**Comment by RangerMauve** [comment]
Is gtk4 still unstable?

**Comment by actionless** [comment]
i have only few gtk4 apps, easyeffects and pavucontrol, and it works ok with those apps since gtk4-oodwaita was created and til now. idk how it behave in other apps

**Comment by actionless** [comment]
or if you mean api of gtk4 - the api of gtk4 would became stable after they'll stop making new releases and start releasing gtk5

**Comment by actionless** [comment]
i've updated my system, and easyeffects moved on from gtk to qt, very telling about current state of gtk

**Comment by RangerMauve** [comment]
Is the `gtk4-oodwaita` theme part of oomox now?

**Comment by actionless** [comment]
of `themix`
